DeepDive.md

```diff
@@ -1,4 +1,4 @@
-# Deep Dive: [Ethereum RPC](https://github.com/internet-computer-protocol/ic-eth-rpc)
+# Deep Dive: [EVM RPC](https://github.com/internet-computer-protocol/ic-eth-rpc)
```

Comment:
The repository should also be renamed to `ic-evm-rpc` (but not necessarily now).